plugin/core/open.py

```diff
@@ -2,17 +2,69 @@
 from .promise import Promise
 from .promise import ResolveFunc
 from .protocol import DocumentUri
+from .protocol import UINT_MAX
 from .protocol import Range
 from .protocol import RangeLsp
 from .typing import Dict, Tuple, Optional
 from .url import parse_uri
 from .views import range_to_region
+from urllib.parse import unquote, urlparse
 import os
+import re
 import sublime
 import subprocess
+import webbrowser
 
 
 opening_files = {}  # type: Dict[str, Tuple[Promise[Optional[sublime.View]], ResolveFunc[Optional[sublime.View]]]]
+FRAGMENT_PATTERN = re.compile(r'^L?(\d+)(?:,(\d+))?(?:-L?(\d+)(?:,(\d+))?)?')
+
+
+def open_file_uri(
+    window: sublime.Window, uri: DocumentUri, flags: int = 0, group: int = -1
+) -> Promise[Optional[sublime.View]]:
+
+    def parse_fragment(fragment: str) -> RangeLsp:
+        match = FRAGMENT_PATTERN.match(fragment)
+        if match:
+            # assume that line and column numbers in the fragment are 1-based
+            start_line, start_column, end_line, end_column = [max(1, int(g)) if g else None for g in match.groups()]
+            if start_line is not None:
+                if end_line is not None:
+                    if start_column is not None and end_column is not None:
+                        return {
+                            "start": {"line": start_line - 1, "character": start_column - 1},
+                            "end": {"line": end_line - 1, "character": end_column - 1}
+                        }
```

Comment:
`Range` has a method `to_lsp()` which would produce the correct type for `center_selection`. But I'm not really convinced that it would look better that way. We would basically create two new Point objects and a Range object, just to immediately convert it into a RangeLsp dict again. Seems a bit inefficient and needless to me.

But in general, if I were asked to redesign the type definitions, I would rename `RangeLsp` to `Range`, remove the current `Range` class altogether, and where necessary use `sublime.Region` together with correspondingly updated functions `region_to_range` and `range_to_region` which will then operate on LSP's "Range" dicts. We usually deal with `sublime.Region`s anyway when when we interact with the View. And currently we always have to use that additional and seemingly unnecessary step to create a `Range` object together with `Range.from_lsp(...)` or `to_lsp()` when converting from or to LSP's data structures.